src/pkg/backup/details/details.go

```diff
@@ -470,11 +483,35 @@ func (d *Details) add(
 		entry.ItemRef = withoutMetadataSuffix(entry.ItemRef)
 	}
 
+	if d.repoRefToEntry == nil {
+		d.repoRefToEntry = make(map[string]*Entry)
+	}
+
+	d.repoRefToEntry[entry.RepoRef] = &entry
 	d.Entries = append(d.Entries, entry)
 
 	return entry, nil
 }
 
+func (d *Details) updateEntry(
+	repoRef string,
+	newInfo ItemInfo,
+) error {
+	if d.repoRefToEntry == nil {
+		d.repoRefToEntry = make(map[string]*Entry)
+	}
+
+	entry, ok := d.repoRefToEntry[repoRef]
+	if !ok {
+		return clues.New("entry not found").With("repo_ref", repoRef)
```

Comment:
nit: use PII safe logging for RepoRef please
```suggestion
		return clues.New("entry not found").With("repo_ref", path.LoggableDir(repoRef))
```